packages/aws-cdk-lib/aws-ecs/lib/base/task-definition.ts

```diff
@@ -1234,3 +1234,29 @@ export function isFargateCompatible(compatibility: Compatibility): boolean {
 export function isExternalCompatible(compatibility: Compatibility): boolean {
   return [Compatibility.EXTERNAL].includes(compatibility);
 }
+
+/**
+ * Represents revision of a task definition, either a specific numbered revision or
+ * the specia "latest" revision
+ */
+export class TaskDefinitionRevision {
+  /**
+   * The most recent revision of a task
+   */
+  static latest() {
+    return new TaskDefinitionRevision('latest');
+  }
+
+  /**
+   * A specfic numbered revision of a task
+   */
+  static revision(revision: number) {
+    return new TaskDefinitionRevision(revision.toString());
+  }
+
+  public readonly revisionId: string;
+
+  constructor(revision: string) {
+    this.revisionId = revision;
+  }
+}
```

Comment:
```suggestion

/**
 * Represents a task definition revision.
 */
export class TaskDefinitionRevision {
  /**
   * The most recent revision of a task
   */
  public static readonly LATEST = new TaskDefinitionRevision('latest');

  /**
   * Specific revision of a task
   */
  public static of(revision: number) {
    if (revision < 1) {
      throw new Error(`A task definition revision must be 'latest' or a positive number, got ${revision}`);
    }
    return new TaskDefinitionRevision(revision.toString());
  }

  private constructor(public readonly revision: string) {}
}
```
Let's follow the enum-class pattern for this.
Thanks for the clarification on union types 👍
A unit test is needed on revision number validation.